src/components/Hooks/useFetchContacts.js

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react'
+import { useQuery } from 'cozy-client'
+
+/**
+ * Hook to fetch contacts according to a query.
+ * FetchMore is triggered as many as necessary to get all contacts
+ * @param {object} query - Definition created with Q()
+ * @returns {array} Query's status {bool} and contacts {array}
+ */
+const useFetchContacts = query => {
+  const [isFetchFinished, setIsFetchFinished] = useState(false)
+  const [contacts, setContacts] = useState([])
+  const { data, fetchStatus, hasMore, fetchMore } = useQuery(
+    query.definition,
+    query.options
+  )
+
+  if (fetchStatus === 'loaded' && !hasMore) {
+    if (isFetchFinished) return [isFetchFinished, contacts]
+    setContacts(data)
+    setIsFetchFinished(true)
```

Comment:
Why do we need to have a `isFetchFinished` state? 


```
if (fetchStatus === 'loaded' && !hasMore) {
setContacts(data)
return [true, contacts]
}
```

Is not fine?

Reply:
No because `fetchStatus === 'loaded' && !hasMore`is the final state and don't change. So the condition is always true at the end. And `setContacts(data)` reload the component, so it will create an infinite loop.

But actually we can simplify more, thanks to you :)